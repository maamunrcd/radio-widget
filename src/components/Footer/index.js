import React from "react";
import "./index.scss";
import { connect } from "react-redux";

const Footer = ({ selectItem }) => {
  return (
    <div className="footer" data-testid="footer">
      {selectItem && selectItem.length > 0
        ? selectItem.map((station, key) => (
            <div key={key}>
              <span className="label">CURRENTLY PLAYING</span>
              <p className="title">{station.title}</p>
            </div>
          ))
        : null}
    </div>
  );
};

const mapStateToProps = (state) => ({
  selectItem: state.selectStation,
});

export default connect(mapStateToProps)(Footer);
