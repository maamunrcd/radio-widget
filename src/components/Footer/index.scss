@import '../../assets/scss/variable.scss';
.footer {
    height: 20%;
    min-height: 20%;
    border-bottom-left-radius: 45px;
    border-bottom-right-radius: 45px;
    background:darken($bg-color, 10%);
    padding: 40px;
    box-sizing: border-box;
    display: flex;
    flex-direction: column;
    align-items:center;
    justify-content: center;
    .label {
        font-size: 22px;
        color: $header-color;
    }
    .title {
        line-height: 1;
        padding: 0;
        margin: 0;
        text-transform: none;
    }
}