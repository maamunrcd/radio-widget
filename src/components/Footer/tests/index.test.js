import React from "react";
import { Provider } from "react-redux";
import { render, screen } from "@testing-library/react";
import Footer from "../index";
import { createStore, applyMiddleware } from "redux";
import createSagaMiddleware from "redux-saga";
import radioStation from "../../../reducers";
import rootSaga from "../../../sagas";
const sagaMiddleware = createSagaMiddleware();

describe("<Footer />", () => {
  let store;
  beforeAll(() => {
    store = createStore(radioStation, applyMiddleware(sagaMiddleware));
    sagaMiddleware.run(rootSaga);
  });
  it("should render its all necessary node", () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <Footer selectItem={null} />
      </Provider>
    );

    expect(firstChild).toMatchSnapshot();
  });

  it("Should show empty div if selectItem props empty", () => {
    render(
      <Provider store={store}>
        <Footer selectItem={null} />
      </Provider>
    );
    const footerNode = screen.getByTestId("footer");
    expect(footerNode.children).toHaveLength(0);
  });

  it("Should show children if selectItem props not empty", () => {
    render(
      <Provider store={store}>
        <Footer
          selectItem={[
            {
              id: 2,
              title: "Dribbble FM",
              freq: "101,2",
            },
          ]}
        />
      </Provider>
    );
    const footerNode = screen.getByTestId("footer");

    expect(footerNode.children).not.toBeNull();
  });
});
