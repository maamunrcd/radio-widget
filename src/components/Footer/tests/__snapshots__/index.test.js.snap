// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render its all necessary node 1`] = `
<div
  class="footer"
  data-testid="footer"
/>
`;
